### DISCUSSION TITLE
移动端语言切换,能否支持RTL布局处理.

### DISCUSSION THREAD

**Discussion by Pacyx** [Tencent/libpag#476] [Ideas] [answered]
移动端会切换阿拉伯语,pag文件特效是否能支持RTL.

**Comment by domchen** [comment, marked as answer]
这个目前还没有支持，预计文本的国际化要很久，但其实目前是有办法实现的。你可以在PAG文件里需要替换文本的地方改成放一个占位图，然后在外部把特殊排版的文本用系统API画到一个图片上，替换到PAG里的占位图上就可以了。

**Reply by jifang** [reply]
三年了，有考虑支持吗？

**Reply by kevingpqi123** [reply]
还没有提上日程， 今年的重点是适配 Metal、Vulkan 等后端，以及计划把 ai 引入进来

**Reply by domchen** [reply]
> 三年了，有考虑支持吗？

有的，最近刚刚加上的。Bidi双向文本，分词断行这些高级排版特性都有了。不过是加在下一代围绕PAGX的新格式渲染能力里。后面发布PAG 5.0的时候，新的渲染架构会完全取代老的。

**Comment by Pacyx** [comment]
感谢回复，我们这边采用了对特殊场景下使用的pagview控件做了左右镜像翻转处理。其他必要场景可能需要我们这边做两套pag资源适配。        
            
                
                    
                            
                                
                            
                            
                                471309445
                            
                    
                        
                            
                                    ***@***.***
                            
                        
                
            
        ---- 回复的原邮件 ----发件人Dom ***@***.***>发送日期2022年08月16日 17:19 ***@***.***> ***@***.******@***.***>主题Re: [Tencent/libpag] 移动端语言切换,能否支持RTL布局处理. (Discussion #476)
这个目前还没有支持，预计文本的国际化要很久，但其实目前是有办法实现的。你可以在PAG文件里需要替换文本的地方改成放一个占位图，然后在外部把特殊排版的文本用系统API画到一个图片上，替换到PAG里的占位图上就可以了。

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you authored the thread.Message ID: ***@***.***>
[
{
***@***.***": "http://schema.org",
***@***.***": "EmailMessage",
"potentialAction": {
***@***.***": "ViewAction",
"target": "https://github.com/Tencent/libpag/discussions/476#discussioncomment-3404890",
"url": "https://github.com/Tencent/libpag/discussions/476#discussioncomment-3404890",
"name": "View Discussion"
},
"description": "View this Discussion on GitHub",
"publisher": {
***@***.***": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]
